text/0000-dynamic-attributes.md

```diff
@@ -0,0 +1,91 @@
+---
+title: Passing Dynamic Attributes/Properties
+status: DRAFTED
+created_at: 2021-07-05
+updated_at: YYYY-MM-DD
+champion: Mohammed Abdul Sattar (abdulsattar)
+pr: https://github.com/salesforce/lwc-rfcs/pull/52
+---
+
+# Passing Dynamic Attributes/Properties
+
+## Summary
+This proposal adds mechanism to pass a dynamic attributes or properties bag to elements in
+an LWC template.
+
+## Example
+```js
+// x/myComponent.js
+export default class MyComponent extends LightningElement {
+    childProps = { 
+        name: 'My Child',
+        style: 'width: 50px',
+        class: 'my-class',
+        onclick: this.handleClick,
+    };
+    
+    handleClick() {
+        console.log('Clicked')
+    }
+}
+```
+```html
+// x/myComponent.html
+<template>
+    <x-child lwc:spread={childProps}></x-child>
+</template>
+```
+
+The new `lwc:spread` directive passes `childProps` as properties/attributes to `x-child`.
+
+## Motivation
+
+In LWC, there is no way to pass props/attributes to a child component when the component author
```

Comment:
this statement is incorrect. It is perfectly possible to do it, but programatically. So, if you want to update this to say that in LWC Templates, then sure... but it is not really something that LWC itself prevents, but the HTML language itself doesn't have that capability.

Reply:
> So, if you want to update this to say that in LWC Templates, then sure... but it is not really something that LWC itself prevents, but the HTML language itself doesn't have that capability.

While I agree with your statement, by going the route of programmatically creating a component, setting the props, and inserting it into the DOM you are automatically opting out of the LWC reactivity and rendering model.

One of the main issues with this approach is that it forces developers to bridge LWC reactivity model with the programmatically created component. Another downside is that it makes the component unable to be server-side renderable.

This proposal might not be the best approach to solve this issue, however, I do believe LWC should provide some new primitives. I will add a point for the next dev sync to make sure we talk through this.